src/ex49_archimedean_spiral.clj

```diff
@@ -1,39 +1,41 @@
 (ns ex49-archimedean_spiral
-  "Generates an archimedean spiral"
+  "Generates an archimedean spiral with an oscillating radius"
   (:require [clojure2d.core :refer :all]
-            [fastmath.core :as m])
-    (:import [fastmath.vector Vec2]))
+            [fastmath.core :as m]
+            [fastmath.vector :as v]))
 
 ;; be sure everything is fast as possible
 (set! *warn-on-reflection* true)
 (set! *unchecked-math* :warn-on-boxed)
 (m/use-primitive-operators)
 
-(def ^:const ^long fps 60)
-
 (def ^:const ^long w 800)
 (def ^:const ^long h 800)
 
 (def ^:const ^double x0 400)
 (def ^:const ^double y0 400)
 
-(def ^:const ^double tStep 0.5)
-
-(def ^:const ^double tAngleScale (/ m/PI 2))
-(def ^:const ^double tRadiusScale 10)
-
-(defn draw [canvas window ^long frame _]
-    (let [field (get-state window)
-          t     ^double (/ frame fps)]
-      (-> canvas
-          (set-background :black 50)
-          (set-color :white 80))
-      (let [p (for [^double theta (range 0 t tStep)] (Vec2.
-                  (+ x0 (* (m/cos (* theta ^double tAngleScale)) (* theta tRadiusScale)))
-                  (+ y0 (* (m/sin (* theta ^double tAngleScale)) (* theta tRadiusScale)))))]
-        (path-bezier canvas p false true))))
-
-(def window (show-window {:canvas (canvas w h)
-                          :window-name "Archimedean spiral"
-                          :draw-fn draw
-                          :fps fps}))
\ No newline at end of file
+(def ^:const ^double tStep 0.01)
+
+(def ^:const ^double tAngleScale m/HALF_PI)
+(def ^:const ^double tRadiusScale 10.0)
+
+(def ^:const ^double ampFactor 0.3)
+(def ^:const ^double periodFactor 16)
+
+(defn draw [canvas window ^double frame _]
+    (let [t1 ^double (/ frame 60.0)]
+        (-> canvas
+            (set-background :black 50)
+            (set-color :white 80))
+        (let [p (for [^double t (range 0.0 t1 tStep)]
+                  (let [theta (* t tAngleScale)
+                        r (* (+ 1 (* ampFactor (m/sin (* periodFactor theta)))) (* t tRadiusScale))]
+                    (vector (+ x0 (* (m/cos theta) r))
```

Comment:
maybe just use `[...]` or `(v/vec2 ...)` instead of explicit `vector`

Reply:
Cool, done